### PR TITLE
Add retrieval activity logs

### PR DESCRIPTION
**Pull request by juliangruber**
To help Station operators understand what's going on

### CODE REVIEW

**Comment by juliangruber** [comment]
https://github.com/filecoin-station/spark/releases/tag/v1.6.0-rc.1

**Comment by juliangruber** [comment]
https://github.com/filecoin-station/core/pull/266

File: lib/spark.js [hunk 1]
```diff
@@ -149,6 +150,16 @@ export default class Spark {
       console.error(err)
     }
 
+    if (stats.timeout) {
+      Zinnia.activity.error('Retrieval timed out')
+    } else if (stats.carTooLarge) {
+      Zinnia.activity.error('CAR too large')
+    } else if (stats.statusCode >= 400) {
+      Zinnia.activity.error('Bad response')
```
Comment (bajtos) [inline lib/spark.js:158]:
```suggestion
      Zinnia.activity.error('Retrieval failed')
```

File: lib/spark.js [hunk 2]
```diff
@@ -149,6 +150,16 @@ export default class Spark {
...
+    } else {
+      Zinnia.activity.info('Retrieval successful')
+    }
+
     const measurementId = await this.submitMeasurement(retrieval, { ...stats })
```
Comment (bajtos) [inline lib/spark.js:163]: Let's report an activity for submitting the measurement, too. `submitMeasurement` throws when the HTTP request fails.

https://github.com/filecoin-station/spark/blob/4c27d6b72982acc02ec3a98fd6fe04b23b9e7cc7/lib/spark.js#L121

File: lib/spark.js (hunk 1 above)
Comment (bajtos) [reply lib/spark.js:158]: The response itself is not necessarily bad. Maybe we are using a protocol that the SP does not support (e.g. bitswap), or there was a timeout on Lassie side while waiting for the next bitswap block, etc.

**Review by bajtos — APPROVED** [review]
LGTM; with a few suggestions that can be applied in a follow-up PR if you like.

File: lib/spark.js (hunk 1 above)
Comment (juliangruber) [reply lib/spark.js:158]: Hmm, you're right that the response itself isn't bad - but it's one where we can't make a retrieval. "Retrieval failed" doesn't tell you anything about what happened however. I would like to find a message that tells us that the SP is not able to serve the retrieval. What about `Unexpected response`?

Comment (bajtos) [reply lib/spark.js:166]: I see. I think you already wrote the message in your text:

```
The Storage Provider is not able to serve the retrieval.
```

File: lib/spark.js [hunk 3]
```diff
@@ -118,9 +118,17 @@ export default class Spark {
...
+    let id
+    try {
+      await assertOkResponse(res, 'Failed to submit measurement')
+      ;({ id } = await res.json())
+      console.log('Measurement submitted (id: %s)', id)
+      Zinnia.activity.info('Measurement submitted')
+    } finally {
+      Zinnia.activity.error('Failed to submit measurement')
+    }
+
```
Comment (bajtos) [inline lib/spark.js:131]: This does not belong to the `finally` block, but instead to `catch`, right? In your proposal, we print this error every time we submit a measurement.

Also, maybe we can simplify the code a bit - I am not a big fan of `let id` + `;({ id } = `

```js
    try {
      await assertOkResponse(res, 'Failed to submit measurement')
      const { id } = await res.json()
      console.log('Measurement submitted (id: %s)', id)
      Zinnia.activity.info('Measurement submitted')
      return id
    } catch (err) {
      Zinnia.activity.error('Failed to submit measurement')
      throw err
    }
```

(this is subjective though)

**Comment by bajtos** [comment]
On the second though, I am not sure if this PR is a good idea. I think we will overload the user with too many detailed messages that will make it difficult to notice the important ones.

If we want to show progress, then aggregated information printed once every X seconds is a better approach, IMO. We are already discussing new activity logs here: https://github.com/filecoin-station/desktop/issues/1123

To allow our users to troubleshoot Station behaviour in details, the current "Save Module Logs as" approach is IMO better. We can review what messages is SPARK printing to this log file via `console.log` and `console.error` APIs and add more logging where needed.

Thoughts?

/cc @patrickwoodhead

**Comment by juliangruber** [comment]
I agree that showing summaries is more effective from a data perspective than reporting any individual retrieval. Let's revisit in our call

**Pull request closed without merging**